src/dmd/parse.d

```diff
@@ -7270,11 +7270,11 @@ final class Parser(AST) : Lexer
             {
                 const(char)* srcfile = mod.srcfile.name.toChars();
                 const(char)* s;
-                if(loc.filename && !FileName.equals(loc.filename, srcfile)) {
-                    s = loc.filename;
-                } else {
+                if (loc.isValid())
+                    s = FileName.toAbsolute(loc.filename);
+                else
```

Comment:
srcfile only used in this branch so should be declared here

Reply:
it's used on line 7273